jaxrs/server-adapters/resteasy-netty4/src/main/java/org/jboss/resteasy/plugins/server/netty/NettyUtil.java

```diff
@@ -31,9 +29,16 @@ public static ResteasyUriInfo extractUriInfo(HttpRequest request, String context
       String host = HttpHeaders.getHost(request, "unknown");
       String uri = request.getUri();
 
-      String uriString = protocol + "://" + host + uri;
+      String uriString;
+
+      // If we appear to have an absolute URL, don't try to recreate it from the host and request line.
+      if (uri.startsWith(protocol + "://")) {
+         uriString = uri;
+      } else {
+         uriString = protocol + "://" + host + uri;
+      }
+
       URI absoluteURI = URI.create(uriString);
-      URI noQuery = UriBuilder.fromUri(uriString).replaceQuery(null).build();
```

Comment:
Glad to help. And very happy to see you working through the PR backlog on this project!